src/r_things.c

```diff
@@ -1053,15 +1086,82 @@ void R_DrawMasked(void)
 {
   int i;
   drawseg_t *ds;
+  const int cx = MAX_SCREENWIDTH / 2;
```

Comment:
Really MAX_SCREENWIDTH here and not currently used screenwidth?

Reply:
But what is the name of currently used screenwidth variable?

Reply:
Could you use `centery` from `r_main.c`?

Reply:
Two moments:

1) `Z_Realloc` should it be...?
```
        drawsegs_xranges[i].items = Z_Realloc(
          drawsegs_xranges[i].items,
          drawsegs_xrange_size * sizeof(drawsegs_xranges[i].items[0]), PU_STATIC, 0); // static? or cache?
```

2) Using `const int cx = centerx / 2` is working, but some reason, I've got about **5** fps less than when using `MAX_SCREENWIDTH`. Really no idea why. @rfomin could you please check this difference on your side?

Reply:
1. When replacing stdlib functions always use`PU_STATIC`, please. 
2. `centerx` is already half screenwidth. :grin:

Reply:
🤤... Pardon me, just figured it out.

Reply:
And done!